scripts/upload_image_dir.py

```diff
@@ -1,97 +1,101 @@
-from pocs.utils import current_time
-from pocs.utils.config import load_config
-from pocs.utils.images import upload_observation_dir
-
-
-def main(date, auto_confirm=False, verbose=False):
-    """Upload images more recent than given date
+from glob import glob
+from astropy.io import fits
 
-    Args:
-        date (datetime): Images more recent than this date will be uploaded
-        auto_confirm (bool, optional): If upload should first be confirm. Default
-            to False.
-        verbose (bool, optional): Verbose output. Default False.
-
-    Returns:
-        int: Number of images uploaded
+from pocs.utils.config import load_config
+from pocs.utils.images import clean_observation_dir
+from pocs.utils.google.storage import upload_observation_to_bucket
+from pocs.utils.db.postgres import get_db_proxy_conn
+from pocs.utils.db.postgres import add_header_to_db
+from pocs.utils import error
+
+
+def main(directory,
+         upload=False,
+         remove_jpgs=False,
+         overwrite=False,
+         make_timelapse=False,
+         send_headers=False,
+         db_pass=None,
+         verbose=False,
+         **kwargs):
+    """Upload images from the given directory.
+
+    See argparse help string below for details about parameters.
     """
-    db = PanMongo()
-    config = load_config()
-    try:
-        pan_id = config['pan_id']
-    except KeyError:
-        warn("Can't upload without a valid pan_id in the config")
-        return
 
     def _print(msg):
         if verbose:
             print(msg)
 
-    img_dir = config['directories']['images']
-    fields_dir = os.path.join(img_dir, 'fields')
-
-    # Get all the sequences from previous day
-    seq_ids = db.observations.distinct(
-        'sequence_id', {'date': {'$gte': date}})
-    # Find all images corresponding to those sequences
-    imgs = [record['data']['file_path'] for record in db.observations.find(
-        {'sequence_id': {'$in': seq_ids}}, {'data.file_path': 1})]
-
-    # Get directory names without leading fields_dir and trailing image dir
-    dirs = sorted(set([img[0:img.rindex('/') - 1].replace(fields_dir, '') for img in imgs]))
-
-    if auto_confirm is False:
-        _print("Found the following dirs for {}:".format(date))
-        pprint(dirs)
-        if input("Proceed (Y/n): ") == 'n':
-            return
-
-    if verbose:
-        dir_iterator = console.ProgressBar(dirs)
-    else:
-        dir_iterator = dirs
-
-    for d in dir_iterator:
-        dir_name = '{}/{}'.format(fields_dir, d)
-
-        upload_observation_dir(pan_id, dir_name)
+    config = load_config()
+    try:
+        pan_id = config['pan_id']
+    except KeyError:
+        raise error.GoogleCloudError("Can't upload without a valid pan_id in the config")
 
-    return len(imgs)
+    _print("Cleaning observation directory: {}".format(directory))
+    try:
+        clean_observation_dir(directory, remove_jpgs=remove_jpgs, include_timelapse=make_timelapse,
```

Comment:
Interesting order of cleanup and upload. I suppose we could do with choices such as:
- Create timelapse
- Create jpg
- Upload fits
- Upload jpg
- Upload timelapse
- Extra upload pattern/files
- Exclude upload pattern/files
- Remove jpg
- Remove timelapse
- Remove fits
- Remove extra pattern/files
- Remove empty directory

And so on. For super small disk systems, we'd want the ability to discard the fits file as soon as it was used to generate the jpg and had been uploaded.

I'm not really asking for anything at this time. Just a bit worried. Maybe we simply need to make sure that we create a very clean separation between the robotic telescope (capturing of data, tracking the sky) and the data processing/upload system.

Reply:
Yes, I will need to make something some time regarding FITS storage as the Pi doesn't have a lot.

Reply:
I've been assuming that a Pi will either require high-speed internet access or a local disk (USB connected). On possibility for a Pi system is to omit creating JPGs and timelapse files locally, and only do that in the cloud. Requires a strong separation between PAWS (including JPG and timelapse creation) and POCS (operating the observatory, without a UI, just an API, maybe a CLI).